server/Arcadia.Assistant.UserPreferences/UserPreferencesActor.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Arcadia.Assistant.UserPreferences
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Akka.Actor;
+    using Akka.Persistence;
+
+    public class UserPreferencesActor : UntypedPersistentActor
+    {
+        private readonly Dictionary<string, UserPreferences> userPreferencesById = new Dictionary<string, UserPreferences>();
+
+        public override string PersistenceId => "user-preferences";
+
+        protected override void OnCommand(object message)
+        {
+            switch (message)
+            {
+                case GetUserPreferencesMessage msg:
+                    var userPreferences = this.userPreferencesById.ContainsKey(msg.UserId)
+                        ? this.userPreferencesById[msg.UserId]
+                        : null;
+                    this.Sender.Tell(new GetUserPreferencesMessage.Response(userPreferences));
+                    break;
+
+                case SaveUserPreferencesMessage msg:
+                    try
+                    {
+                        this.Persist(msg, m =>
```

Comment:
you never persist a message. Messages define actor contract, it might change (one event can cause multiple events to be fired and so on). What you want to do is to persist events - 'UserChangesPreferences' event, for instance. Also, you can split this event on multiple types - 'UserEnablesEmailNotifications', 'UserEnablesPushNotifications' and handle them differently. For example, when you have devices stored, you might want to remove them from the dictionary of recipients when you handle 'userDisablesPushNotifications' event.

Reply:
Done.